src/tests/test_split.cpp

```diff
@@ -0,0 +1,150 @@
+#include <filesystem>
+#include <list>
+#include <set>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algebraic_constraints/ac_algorithm.h"
+#include "algorithms/algebraic_constraints/bin_operation_enum.h"
+#include "algorithms/algo_factory.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+#include "types.h"
+
+namespace tests {
+
+void CompareDDStringLists(std::set<std::pair<std::set<model::DFStringConstraint>,
+                                             std::set<model::DFStringConstraint>>> const& expected,
+                          std::list<model::DDString> const& actual) {
+    ASSERT_EQ(expected.size(), actual.size())
+            << "count of generated dependencies does not match: expected " << expected.size()
+            << ", got " << actual.size();
+    for (auto const& dd : actual) {
+        std::set<model::DFStringConstraint> lhs(dd.left.begin(), dd.left.end());
+        std::set<model::DFStringConstraint> rhs(dd.right.begin(), dd.right.end());
+        if (expected.find(std::make_pair(std::move(lhs), std::move(rhs))) == expected.end()) {
+            FAIL() << "generated dependency that is not expected";
+        }
+    }
+}
+
+class SplitAlgorithmTest : public ::testing::Test {
+public:
+    static algos::StdParamsMap GetParamMap(CSVConfig const& csv_config,
+                                           std::filesystem::path const& dif_table_path,
+                                           char dif_table_separator, bool dif_table_has_header) {
+        using namespace config::names;
+        if (dif_table_path == "") {
+            return {{kCsvConfig, csv_config}};
+        }
+        config::InputTable table = std::make_shared<CSVParser>(dif_table_path, dif_table_separator,
+                                                               dif_table_has_header);
+        return {{kCsvConfig, csv_config}, {kDifferenceTable, std::move(table)}};
+    }
+
+    static std::unique_ptr<algos::dd::Split> CreateSplitAlgorithmInstance(
+            CSVConfig const& csv_config, std::filesystem::path const& dif_table_path = "",
+            char dif_table_separator = ',', bool dif_table_has_header = true) {
+        return algos::CreateAndLoadAlgorithm<algos::dd::Split>(
+                GetParamMap(csv_config, dif_table_path, dif_table_separator, dif_table_has_header));
+    }
+};
+
+TEST_F(SplitAlgorithmTest, Test0) {
+    auto algo = CreateSplitAlgorithmInstance(
+            kTestDD, std::filesystem::current_path() /
+                             "../../src/core/algorithms/dd/split/datasets/TestDif.csv");
```

Comment:
Just `src/core/algorithms/dd/split/datasets/TestDif.csv` without prefix does not work?